summingbird-storm/src/main/scala/com/twitter/summingbird/storm/ComponentProvider.scala

```diff
@@ -0,0 +1,32 @@
+package com.twitter.summingbird.storm
+
+import com.twitter.algebird.Semigroup
+import com.twitter.summingbird.batch.Batcher
+import com.twitter.summingbird.online.executor.KeyValueShards
+import com.twitter.summingbird.storm.builder.Topology
+
+/**
+ * This trait describes how to build topology component which corresponds to `Producer`s chain.
```

Comment:
It's not clear why ComponentProvider needs to implement two methods below and what each out is required for at the conceptual level.